src/Mod/Sketcher/Gui/EditDatumDialog.cpp

```diff
@@ -201,7 +201,8 @@ int EditDatumDialog::exec(bool atCursor)
 void EditDatumDialog::accepted()
 {
     Base::Quantity newQuant = ui_ins_datum->labelEdit->value();
-    if (newQuant.isQuantity() || (Constr->Type == Sketcher::SnellsLaw && newQuant.isDimensionless())
+    if (!newQuant.isDimensionless()
+        || (Constr->Type == Sketcher::SnellsLaw && newQuant.isDimensionless())
```

Comment:
`isQuantity` is changed to `isDimensionless` here which makes the expression a bit strange compared to the old one.

As we're already checking if `!newQuant.isDimensionless()` this can be simplified to:
```cpp
    if (!newQuant.isDimensionless()
        || Constr->Type == Sketcher::SnellsLaw
        || Constr->Type == Sketcher::Weight) {
```

I don't know if that is correct (it does look sound), but maybe you could double check it as you have better domain knowledge?